app/src/contexts/surveyContext.tsx

```diff
@@ -0,0 +1,84 @@
+import { useBiohubApi } from 'hooks/useBioHubApi';
+import useDataLoader, { DataLoader } from 'hooks/useDataLoader';
+import { IGetSurveyForViewResponse } from 'interfaces/useSurveyApi.interface';
+import React, { createContext, PropsWithChildren, useEffect, useMemo } from 'react';
+import { useParams } from 'react-router';
+
+/**
+ * Context object that stores information about a survey
+ *
+ * @export
+ * @interface ISurveyContext
+ */
+export interface ISurveyContext {
+  /**
+   * The Data Loader used to load survey data
+   *
+   * @type {DataLoader<[project_id: number, survey_id: number], IGetSurveyForViewResponse, unknown>}
+   * @memberof ISurveyContext
+   */
+  surveyDataLoader: DataLoader<[project_id: number, survey_id: number], IGetSurveyForViewResponse, unknown>;
+
+  /**
+   * The project ID belonging to the current survey
+   *
+   * @type {number}
+   * @memberof ISurveyContext
+   */
+  projectId: number;
+
+  /**
+   * The ID belonging to the current survey
+   *
+   * @type {number}
+   * @memberof ISurveyContext
+   */
+  surveyId: number;
+}
+
+export const SurveyContext = createContext<ISurveyContext>({
+  surveyDataLoader: {} as DataLoader<[project_id: number, survey_id: number], IGetSurveyForViewResponse, unknown>,
+  projectId: -1,
+  surveyId: -1
+});
+
+export const SurveyContextProvider = (props: PropsWithChildren<Record<never, any>>) => {
```

Comment:
Question:  Is `props: PropsWithChildren<Record<never, any>>` still a good idea?  I wonder if we should specify the props that the context is using.... or maybe even go with React.PropsWithChildren<{}>

Reply:
See: https://stackoverflow.com/a/66271367/2932330

> I wonder if we should specify the props that the context is using
Since the Provider doesn't take any props (except children), we wouldn't want to type it with `ISurveyContext`.

> or maybe even go with React.PropsWithChildren<{}>
I tried this, but according the ES lint there are situations where this confuses the compiler.

Reply:
Or at least, something like that, I cant remember the specifics anymore

Reply:
And then `object` and `Object` also factor in somewhere... but I think `Record` is still preferred in this case.